File: mobile/src/wirelessGecko/java/com/siliconlabs/bledemo/rangetest/TxPower.java
package com.siliconlabs.bledemo.rangetest;

import android.support.annotation.NonNull;

/**
 * @author Comarch S.A.
 */

public class TxPower {

    private final int characteristicValue;

    public TxPower(int characteristicValue) {
        this.characteristicValue = characteristicValue;
    }

    public int asCharacteristicValue() {
        return characteristicValue;
    }

    public float asDisplayValue() {
        return characteristicValue / 10f;
    }

    @NonNull
    @Override
    public String toString() {
        return String.valueOf(asDisplayValue());
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TxPower txPower = (TxPower) o;

        return characteristicValue == txPower.characteristicValue;
    }

    @Override
    public int hashCode() {
        return characteristicValue;
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/RangeTest/Models/TxPower.kt
package com.siliconlabs.bledemo.RangeTest.Models

/**
 * @author Comarch S.A.
 */
class TxPower(private val characteristicValue: Int) {
    fun asCharacteristicValue(): Int {
        return characteristicValue
    }

    fun asDisplayValue(): Float {
        return characteristicValue / 10f
    }

    override fun toString(): String {
        return asDisplayValue().toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val txPower = other as TxPower
        return characteristicValue == txPower.characteristicValue
    }

    override fun hashCode(): Int {
        return characteristicValue
    }

}